buildchain/buildchain/docker_command.py

```diff
@@ -22,6 +22,29 @@
 DOCKER_CLIENT : docker.DockerClient = docker.from_env()
 
 
+def task_errors(
```

Comment:
Can be squashed on a single line.

also `expected` instead of `caught` might be a better name, e.g. `def task_errors(*expected_exn: Type[Exception]) -> Callable[[Any], Any]:`

Nice helper BTW, it really makes the code clearer.